hsqldb/src/test/java/org/spine3/server/storage/hsqldb/HsqlEntityStorageShould.java

```diff
@@ -0,0 +1,102 @@
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.storage.hsqldb;
+
+import com.google.protobuf.StringValue;
+import com.zaxxer.hikari.HikariConfig;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.spine3.server.storage.EntityStorageShould;
+
+import static org.junit.Assert.fail;
+import static org.spine3.server.storage.hsqldb.HsqlEntityStorage.*;
+
+/**
+ * @author Alexander Litus
+ */
+@SuppressWarnings("InstanceMethodNamingConvention")
+public class HsqlEntityStorageShould extends EntityStorageShould {
+
+    /**
+     * The URL of in-memory database.
+     */
+    private static final String DB_URL = "jdbc:hsqldb:mem:entitytests";
+
+    private static final HsqlDb DATABASE = newHsqlDb();
+
+    private static final HsqlEntityStorage<String, StringValue> STORAGE = newStorage(DATABASE);
+
+    private static final String CREATE_TABLE_SQL =
+            "CREATE TABLE " + ENTITIES + '(' +
+                ID + " VARCHAR(999)," +
+                ENTITY + " BLOB," +
+                "PRIMARY KEY(" + ID + ')' +
+            ");";
+
+    private static final String DROP_TABLE_SQL = "DROP TABLE " + ENTITIES;
+
+    private static final String SHUTDOWN_SQL = "SHUTDOWN";
+
+    public HsqlEntityStorageShould() {
+        super(STORAGE);
+    }
+
+    private static HsqlDb newHsqlDb() {
+        final HikariConfig config = new HikariConfig();
+        config.setJdbcUrl(DB_URL);
+        return HsqlDb.newInstance(config);
+    }
+
+    private static HsqlEntityStorage<String, StringValue> newStorage(HsqlDb db) {
+        return HsqlEntityStorage.newInstance(db, StringValue.getDescriptor());
+    }
+
+    @Before
+    public void setUpTest() {
+        DATABASE.execute(CREATE_TABLE_SQL);
+    }
+
+    @After
+    public void tearDownTest() {
+        DATABASE.execute(DROP_TABLE_SQL);
+    }
+
+    @AfterClass
+    public static void tearDownClass() {
+        DATABASE.execute(SHUTDOWN_SQL);
+        DATABASE.close();
+    }
+
+    @Test
+    public void close() {
```

Comment:
Why not name it `close_itself()` or `auto_close_`?
The way it's now is a bit confusing. If you overlook `@Test` annotation, which I initially did, the method looks like a real service method of the test case.